### ISSUE TITLE
[java] False positive "UselessParentheses" for parentheses that contain assignment

### ISSUE THREAD

**Issue by cobratbq** [pmd/pmd#1285]
**Affects PMD Version:** PMD 6.5.0, ran through maven-pmd-plugin 3.10.0.

**Rule:** UselessParentheses (category/java/codestyle.xml/UselessParentheses)

**Description:** This detector does not detect the value of the extra parentheses to ensure that the right result is assigned to the in-line assignment.

**Code Sample demonstrating the issue:**

```
Matcher matcher;
if ((matcher = PATTERN_V4.matcher(message)).matches()) {
    ...
}
```

**Running PMD through:** Maven
